server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1256,13 +1256,24 @@ private static boolean isFiltered(Property[] properties) {
     public static class RegexValidator implements Writeable, Validator<String> {
         private Pattern pattern;
 
+        private boolean isMatching = true;
+
         /**
          * @param regex A regular expression containing the only valid input for this setting.
          */
         public RegexValidator(String regex) {
             this.pattern = Pattern.compile(regex);
         }
 
+        /**
+         * @param regex A regular expression containing the only valid input for this setting.
```

Comment:
This sentence is no longer true all the time with the boolean.  Consider adding a summary statement like "Constructs a validator based on a regular expression."  and then this param can just be "the regular expression".